src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/HardCodeEncryptionKeySources.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities.PooledObjects;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    internal static class HardCodeEncryptionKeySources
+    {
+        /// <summary>
+        /// <see cref="SourceInfo"/>s for hardcoded key tainted data sources.
+        /// </summary>
+        public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
+
+        /// <summary>
+        /// Statically constructs.
+        /// </summary>
+        static HardCodeEncryptionKeySources()
+        {
+            var builder = PooledHashSet<SourceInfo>.GetInstance();
+
+            builder.AddSourceInfo(
+                WellKnownTypeNames.SystemConvert,
+                isInterface: false,
+                taintedProperties: null,
+                taintedMethods: new[] {
+                    "FromBase64String",
+                });
```

Comment:
How are we indicating that this method is a tainted data source only if its parameter is a constant value?

Reply:
How about adding a lambda function as a condition check into taintedMethods?

Reply:
Oops, sorry, I forgot to respond yesterday.

I think that's fine.  For lambdas to have access to per compilation data like INamedTypeSymbols, one approach is to put a lambda factories in the SourceInfo (or other infos), to produce the lambdas based on the compilation.

It's adding complexity in TaintedDataConfig, but the flexibility would be nice as we come across other complicated tainted data scenarios.